From 83e4d816e1a854fd6e8f3375e75cff287918f19b Mon Sep 17 00:00:00 2001
From: ejaaskel <esa.jaaskela@suomi24.fi>
Date: Wed, 19 Nov 2025 06:37:27 +0000
Subject: [PATCH] SELinux fixes for Raspberry Pi

Running Raspberry Pi with SELinux requires a few changes to the
refpolicy, as the fstab is unlabeled. Additionally, logrotate wants
access to /dev/ttyS0 and /dev/console.

The fstab becomes unlabeled because a mount point for /boot partition
is defined in the .wks file for Raspberry Pi, and that results in fstab
getting modified. If you want to avoid that, remove the /boot mount
point from the .wks file and the label should stay intact.

Note that multiple logical changes should be in separate patches,
however, in this case all the Raspberry Pi related changes are in a
single patch file.
---
 policy/modules/admin/logrotate.te    | 3 +++
 policy/modules/system/mount.te       | 1 +
 policy/modules/system/selinuxutil.te | 1 +
 3 files changed, 5 insertions(+)

diff --git a/policy/modules/admin/logrotate.te b/policy/modules/admin/logrotate.te
index 287941776..ceebdbaa4 100644
--- a/policy/modules/admin/logrotate.te
+++ b/policy/modules/admin/logrotate.te
@@ -29,6 +29,9 @@ files_type(logrotate_var_lib_t)
 type logrotate_unit_t;
 init_unit_file(logrotate_unit_t)

+gen_require(`
+       type tty_device_t;
+')

 ########################################
 #
@@ -54,6 +57,9 @@ allow logrotate_t self:msg { receive send };
 allow logrotate_t logrotate_lock_t:file manage_file_perms;
 files_lock_filetrans(logrotate_t, logrotate_lock_t, file)
 
+allow logrotate_t tty_device_t:chr_file { read write };
+allow logrotate_t init_t:fd use;
+
 manage_dirs_pattern(logrotate_t, logrotate_tmp_t, logrotate_tmp_t)
 manage_files_pattern(logrotate_t, logrotate_tmp_t, logrotate_tmp_t)
 files_tmp_filetrans(logrotate_t, logrotate_tmp_t, { file dir })
diff --git a/policy/modules/system/mount.te b/policy/modules/system/mount.te
index 3fc37619e..6e5522f35 100644
--- a/policy/modules/system/mount.te
+++ b/policy/modules/system/mount.te
@@ -51,6 +51,7 @@ dontaudit mount_t self:process setrlimit;
 
 allow mount_t mount_tmp_t:file manage_file_perms;
 allow mount_t mount_tmp_t:dir manage_dir_perms;
+allow mount_t unlabeled_t:file { read open getattr };
 
 can_exec(mount_t, mount_exec_t)
 
diff --git a/policy/modules/system/selinuxutil.te b/policy/modules/system/selinuxutil.te
index 88b2980d3..c03d6f3e1 100644
--- a/policy/modules/system/selinuxutil.te
+++ b/policy/modules/system/selinuxutil.te
@@ -633,6 +633,7 @@ allow setfiles_t { policy_src_t policy_config_t file_context_t default_context_t
 allow setfiles_t { policy_src_t policy_config_t file_context_t default_context_t }:file read_file_perms;
 allow setfiles_t { policy_src_t policy_config_t file_context_t default_context_t }:lnk_file { ioctl lock read_lnk_file_perms };
 allow setfiles_t file_context_t:file map;
+allow setfiles_t unlabeled_t:file read;
 
 kernel_read_system_state(setfiles_t)
 kernel_relabelfrom_unlabeled_dirs(setfiles_t)
-- 
2.34.1

